samples/string_interning.cpp

```diff
@@ -0,0 +1,18 @@
+#include <vector>
+#include <string>
+
+char* add_string(char *string, std::vector<char*> &old_strings) {
```

Comment:
- `add_string` is probably not the best name in the world :) It's quite generic, I think something like `intern_string` is better.
- You don't want the data structure that holds the saved strings to be visible to the caller. In this case, `old_strings`. The caller should pass a string and get a string.
- You can try returning and passing `const char *`. It's not that `const` is that useful in C / C++, but it just makes
it somewhat clearer and safer that the strings are _only_ for reading.